app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala

```diff
@@ -43,22 +47,24 @@ object MtdItIdBusinessDetails {
   implicit val reads = Json.reads[MtdItIdBusinessDetails]
 }
 
-case class ItsaRelationship(arn: Arn)
+case class ItsaRelationship(arn: Arn, endDate: LocalDate)
 
 object ItsaRelationship {
   implicit val relationshipWrites = Json.writes[ItsaRelationship]
 
-  implicit val reads: Reads[ItsaRelationship] =
-    (JsPath \ "agentReferenceNumber").read[Arn].map(arn => ItsaRelationship(arn))
+  implicit val reads: Reads[ItsaRelationship] = (
+    (JsPath \ "agentReferenceNumber").read[Arn] and
+    (JsPath \ "dateTo").read[LocalDate])(ItsaRelationship.apply _)
 }
 
-case class VatRelationship(arn: Arn)
+case class VatRelationship(arn: Arn, endDate: LocalDate)
```

Comment:
should `endDate` be optional?

Reply:
could be the same argument for ARN and further down the line converts the NotFoundException to None -->. https://github.com/hmrc/agent-client-relationships/blob/master/app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala#L136

Reply:
the endDate default value is 9999-12-31 when creating the relationship, and gets updated when deleted. So I gathered from Paul